### PR TITLE
Bundle Qt and add WebKitGTK

### PR DESCRIPTION
**Pull request by gasinvein**
This allows us to build [2.7.2](https://github.com/flathub/org.telegram.desktop/pull/254)
Upstream always wanted to use their custom-patched Qt anyway.

### CODE REVIEW

**Comment by ilya-fedin** [comment]
I don't feel like we need entire GNOME runtime just for Webkit2GTK. This will produce a lot of questions from people who want to avoid GNOME, tdesktop is a Qt application after all.

File: org.telegram.desktop.json [hunk 1]
```diff
@@ -124,6 +127,122 @@
...
+                "-system-proxies",
+                "-dbus-linked",
+                "-openssl-linked",
+                "-fontconfig",
+                "-accessibility",
+                "-glib",
+                "-icu",
+                "-no-directfb",
+                "-no-linuxfb",
+                "-no-kms"
```
Comment (ilya-fedin) [inline org.telegram.desktop.json:158]: Syscalls from new kernels should be avoided
```suggestion
                "-no-kms",
                "-no-feature-getentropy",
                "-no-feature-renameat2",
                "-no-feature-statx"
```

File: org.telegram.desktop.json [hunk 2]
```diff
@@ -124,6 +125,152 @@
...
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://webkitgtk.org/releases/webkitgtk-2.32.0.tar.xz",
+                    "sha256": "9d7df4dae9ada2394257565acc2a68ace9308c4c61c3fcc00111dc1f11076bf0",
+                    "x-checker-data": {
+                        "type": "html",
+                        "url": "https://webkitgtk.org/releases/",
+                        "version-pattern": "LATEST-STABLE-(\d[\.\d]+\d)",
```
Comment (gasinvein) [inline org.telegram.desktop.json:268]:
```suggestion
                        "version-pattern": "LATEST-STABLE-(\\d[\\.\\d]+\\d)",
```

**Comment by ilya-fedin** [comment]
Looks like it finally built like it should. Let's merge this and put flatpak build in one line with other official builds, providing the same experience, including shadows on Wayland and other Qt patches that increase tdesktop UX.

**Pull request merged**